docs/source/llms/prompt-engineering.rst

```diff
@@ -0,0 +1,351 @@
+button to store this information, along with your choice of LLM, as an MLflow Run. This will
+create a new Run with the prompt template, parameters, and choice of LLM stored as Run params.
+It will also automatically create an MLflow Model with this information that can be used for batch
+or real-time inference.
+
+1. To view this information, click the Run name to open the **Run** page:
+
+    .. figure:: ../_static/images/prompt_eng_run_page.png
+       :scale: 25%
+       :align: center
+
+2. You can also see the parameters and compare them with other configurations by opening the **Table**
+   view tab:
+
+    .. figure:: ../_static/images/prompt_eng_table_view.png
+       :scale: 25%
+       :align: center
+
+3. After your Run is created, MLflow will open the **Evaluation** tab where you can see your latest
+   playground input & output and try out additional inputs:
+
+    .. figure:: ../_static/images/eval_view_1.png
+       :scale: 25%
+       :align: center
+
+
+Step 8: Try new inputs
+----------------------
+To test the behavior of your chosen LLM, prompt template, and parameters on a new inputs:
+
+1. Click the *Add Row* button and fill in a value(s) your prompt template's input variable(s).
+   For example, in the MLflow documentation use case, you can try asking a question
+   unrelated to MLflow to see how the LLM responds. This is important to ensure that the application
+   is robust to irrelevant inputs.
+
+    |add_row| |add_row_modal|
+
+    .. |add_row| image:: ../_static/images/add_row.png
+       :width: 10%
+
+    .. |add_row_modal| image:: ../_static/images/add_row_modal.png
+       :width: 50%
+
+2. Then, click the **Evaluate** button to see the output.
+
+    .. figure:: ../_static/images/evaluate_new_input.png
+        :align: center
+        :scale: 40%
+
+3. Finally, click the **Save** button to store the new inputs and output.
+
+    .. figure:: ../_static/images/save_new_input.png
+        :align: center
+        :scale: 40%
+
+Step 9: Adjust your prompt template and create a new Run
+--------------------------------------------------------
+As you try additional inputs, you might discover scenarios where your choice of LLM, prompt
+template, and parameters doesn't perform as well as you would like. For example, in the
+MLflow documentation use case, the LLM still attempts to answer irrelevant
+questions about :ref:`MLflow Projects <projects>` even if the answer does not appear in the
+specified article.
+
+1. To improve performance, create a new Run by selecting the *Duplicate run* option from the context
+   menu. For example, in the MLflow documentation use case, adding the following text to
+   the prompt template helps improve robustness to irrelevant questions:
+
+   .. code-block::
+
+      If the question does not relate to the article, respond exactly with the phrase
+      "I do not know how to answer that question." Do not include any additional text in your
+      response.
```

Comment:
<img width="1204" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/337f154d-cc2f-4af2-be92-da9767a0a293">

```suggestion
   .. code-block:: text

      If the question does not relate to the article, respond exactly with the phrase
      "I do not know how to answer that question." Do not include any additional text in your
      response.
```

to disable syntax highlighting on non-python code

Reply:
Thank you! Fixed!